Fix overflow scrollbars on Windows for date-range-picker

jeelpansheriya opened the pull request:
Add overflow-auto class to CalendarPrimitive to prevent unnecessary scrollbars on Windows when content fits.

Fixes #122

**Description**

This PR fixes an issue where the date-range-picker (and single date picker) components displayed unnecessary scrollbars on Windows, even when the content did not overflow. The problem was caused by the default `overflow: scroll` behavior in the underlying `react-day-picker` library, which forces scrollbars to appear on Windows.

The fix adds `className="overflow-auto"` to the `CalendarPrimitive` component in both the single and range date picker implementations. This overrides the default overflow behavior, ensuring scrollbars only appear when content actually overflows, improving the user experience on Windows.

**Related issue(s)**

- Fixes #122

**What kind of change does this PR introduce?** (check at least one)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] New Feature (BREAKING CHANGE which adds functionality)
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**How has this been tested?**

- Ran `pnpm lint` on the tremor repository, which passed without errors for the modified file.
- The change is a simple CSS class addition that does not alter functionality, only visual behavior.
- Existing tests for the DatePicker component should continue to pass (no regressions expected).
- Manually verified the code changes in the DatePicker.tsx file.

**Screenshots (if appropriate):**

No screenshots needed, as this is a behavioral fix for scrollbars that may not be visible in all environments. The issue description in #122 includes a screenshot of the problem.

**The PR fulfils these requirements:**

- [x] It's submitted to the `main` branch
- [x] Add refs #XXX or fixes #XXX to the related issue section if your PR refers to or fixes an issue.
- [ ] My change requires a change to the documentation. (Managed by Tremor Team)
- [ ] I have added tests to cover my changes
- [x] Check the ["Allow edits from maintainers" option](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) while creating your PR.
- [x] By contributing to Tremor, you confirm that you have read and agreed to Tremor's [CONTRIBUTING.md](https://github.com/tremorlabs/tremor-npm/blob/main/CONTRIBUTING.md) guideline. You also agree that your contributions will be licensed under the [Apache License 2.0](https://github.com/tremorlabs/tremor?tab=Apache-2.0-1-ov-file#readme) license.

mitrotasios commented:
@Ehesp thanks for the PR! Would you mind elaborating what kind of text you would like to add to the selection?

Ehesp commented:
Well anything I want really. Showing "3 selected" isn't that useful in my UI. Instead I'd like to show what is selected using the values.

mitrotasios commented:
In that case I think something like a list of clickable buttons would be useful. E.g. like this implementation:
<img width="787" alt="image" src="https://user-images.githubusercontent.com/58337217/198875021-60804af6-7ce0-44a6-af2e-90f29b2a9148.png">

Initially we wanted to go for this approach but it turns out to be quite tricky to design such a component fulfilling the responsive criteria for cases where there are many selections (same goes for a simple text input).

I think your point is very valid, but we wouldn't like to rush such decisions as we want to avoid breaking changes in the future. I hope you understand. We will try to figure out a suitable responsive solution and come back to your PR. We would also appreciate any ideas on this :)

Ehesp commented:
That's fine, although this isn't a breaking change since its an optional prop.

But yeah appricate it needs to be considered.

mitrotasios commented:
yeah, I mean a breaking change in the sense that we would need to deprecate that prop if we chose to go with another solution in the future :) Thanks for understanding!

mitrotasios commented:
We will probably not go ahead with this feature at the moment. Sorry about that and thanks for the work!

jeelpansheriya committed:
Fix overflow scrollbars on Windows for date-range-picker

Add overflow-auto class to CalendarPrimitive to prevent unnecessary scrollbars on Windows when content fits.

Fixes #122
